tutorials/nservicebus-sagas/2-timeouts/Snippets/Core_7/BuyersRemorseCancellingOrders/BuyersRemorseCancellingOrders.cs

```diff
@@ -0,0 +1,71 @@
+﻿namespace Core_7.BuyersRemorseCancellingOrders
+{
+    using NServiceBus;
+    using NServiceBus.Logging;
+    using System;
+    using System.Threading.Tasks;
+
+    class UICommands
+    {
+        static ILog log = LogManager.GetLogger<UICommands>();
+
+        async Task Execute(IMessageSession endpointInstance)
+        {
+            #region BuyersRemorseCancellingOrders
+            var lastOrder = string.Empty;
+
+            while (true)
+            {
+                log.Info("Press 'P' to place an order, 'C' to cancel last order, or 'Q' to quit.");
+                var key = Console.ReadKey();
+                Console.WriteLine();
+
+                switch (key.Key)
+                {
+                    case ConsoleKey.P:
+                        // Instantiate the command
+                        var command = new PlaceOrder
+                        {
+                            OrderId = Guid.NewGuid().ToString()
```

Comment:
Never understood why we ToString those :-)

Reply:
Me neither, but to change it we need to change lesson 1 as well. That is out of scope.